File: app/src/test/java/com/example/myapplication/Node.java
package com.example.myapplication;

import java.util.ArrayList;
import java.util.List;

public class Node {
    int value;
    List<Node> nextNodes = new ArrayList<>();

    public Node(int _value) {
        value = _value;
    }

    public void add(Node node) {
        nextNodes.add(node);
    }

    //Node{value=1, nextNodes=}
    // Expected result =
    //[1,2]
}

File: app/src/main/java/Node.kt
import java.util.LinkedList

class Node<T>(
    /**
     * This is the int value of the new Node
     */
    val value: T
) {
    /**
     * NextNodes is a collection of connecting Nodes
     * LinkedList is used to save time complexity of addition of new Node
     * ArrayList has time complexity of O(N) for adding a new Node but LinkedList has time complexity of O(1) for adding a new Node
     */
    private val _nextNodes : MutableList<Node<T>> = LinkedList()
    val nextNodes:List<Node<T>> get() = _nextNodes

    /**
     * Function to add @param node to next nodes
     */
    fun addNode(node: Node<T>) {
        _nextNodes.add(node)
    }

    override fun toString(): String {
        return "Node{" +
                "value=" + value +
                ", nextNodes=" + nextNodes.map { it.value }.joinToString(",") +
                '}'
    }
}